core/amber/src/main/scala/edu/uci/ics/texera/web/resource/dashboard/user/dataset/DatasetResource.scala

```diff
@@ -561,6 +576,54 @@ object DatasetResource {
   case class DatasetNameModification(did: UInteger, name: String)
 
   case class DatasetDescriptionModification(did: UInteger, description: String)
+
+  private def calculateSize(did: UInteger, versionHash: Option[String] = None): Long = {
+    Try {
+      val datasetPath = PathUtils.getDatasetPath(did)
+      val hash = versionHash.getOrElse {
+        fetchLatestDatasetVersionInternal(context, did)
+          .map(_.getVersionHash)
+          .getOrElse(throw new NoSuchElementException("No versions found for this dataset"))
+      }
+
+      val fileNodes = GitVersionControlLocalFileStorage.retrieveRootFileNodesOfVersion(
+        datasetPath,
+        hash
+      )
+
+      calculateSizeFromPhysicalNodes(fileNodes)
+    } match {
+      case Success(size) => size
+      case Failure(exception) =>
+        val errorMessage = versionHash.map(_ => "dataset version").getOrElse("dataset")
+        println(s"Error calculating $errorMessage size: ${exception.getMessage}")
+        0L
+    }
+  }
+
+  def calculateDatasetVersionSize(did: UInteger, dvid: UInteger): Long = {
+    val versionHash = getDatasetVersionByID(context, dvid).getVersionHash
+    calculateSize(did, Some(versionHash))
+  }
+
+  def calculateDatasetSize(did: UInteger): Long = {
+    calculateSize(did)
+  }
+
+  private def calculateSizeFromPhysicalNodes(nodes: java.util.Set[PhysicalFileNode]): Long = {
+    nodes.asScala.foldLeft(0L) { (totalSize, node) =>
+      totalSize + (if (node.isDirectory) {
+                     calculateSizeFromPhysicalNodes(node.getChildren)
+                   } else {
+                     Try(Files.size(node.getAbsolutePath)).getOrElse(0L)
```

Comment:
what if the node doesn't exist in current version? Seems it return 0. I am thinking if we can get the metadata like file size in `JGitVersionControl.getRootFileNodeOfCommit`, when this method is traversing the version's file tree. And we can save this size information as a new member variable of `PhysicalFileNode`, and return that value in this method. Can you look into it? Thanks!

Reply:
Thank you for the suggestions! I didn't know that we had such a function `JGitVersionControl.getRootFileNodeOfCommit`. I'll take a look.

Reply:
I modified the `JGitVersionControl.getRootFileNodeOfCommit` method to include file size information when traversing the file tree of a specific commit. I also added a new member variable to the `PhysicalFileNode` class to store this size information.